app/scripts/modules/core/src/pipeline/config/stages/manualJudgment/ManualJudgmentApproval.tsx

```diff
@@ -29,17 +33,60 @@ export class ManualJudgmentApproval extends React.Component<
       submitting: false,
       judgmentDecision: null,
       judgmentInput: {},
+      applicationRoles: {},
+      userRoles: [],
       error: false,
     };
   }
 
+  public componentDidMount() {
+    const applicationName = this.props.execution.application;
+    ApplicationReader.getApplicationPermissions(applicationName).then(result => {
+      if (typeof result !== 'undefined') {
+        this.setState({
+          applicationRoles: result,
+        });
+        this.isManualJudgmentStageNotAuthorized();
```

Comment:
I am a bit confused as to why this is invoked here. The function returns a boolean, so I would expect that value to be set to the `state` here. 

I suggest (coupled with my recommendations for this functions, to set the authorizations as part of the state and using that value for your conditionals in the html

`this.setState({ authorizedForManualJudgment: this.isUserAuthorizedForManualJudgment() })`